api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/DosageTranslatorImpl.java

```diff
@@ -43,6 +50,32 @@ public Dosage toFhirResource(@Nonnull DrugOrder drugOrder) {
 		dosage.setRoute(conceptTranslator.toFhirResource(drugOrder.getRoute()));
 		dosage.setTiming(timingTranslator.toFhirResource(drugOrder));
 		
+		if (drugOrder.getDose() != null || drugOrder.getDoseUnits() != null) {
+			Dosage.DosageDoseAndRateComponent doseAndRate = new Dosage.DosageDoseAndRateComponent();
+			Quantity dose = new SimpleQuantity();
+			dose.setValue(drugOrder.getDose());
+			if (drugOrder.getDoseUnits() != null) {
+				CodeableConcept doseUnits = conceptTranslator.toFhirResource(drugOrder.getDoseUnits());
+				if (doseUnits != null) {
```

Comment:
Per ticket comments, this will use the RxNorm mapping if present, then the snomed ct if present, and then the uuid with null system otherwise.